sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py

```diff
@@ -469,6 +469,153 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useles
         super().__init__(*args, type="function", **kwargs)
 
 
+class ContentItem(_model_base.Model):
+    """An abstract representation of a structured content item within a chat message.
+
+    You probably want to use the sub-classes and not this class directly. Known sub-classes are:
+    ImageContentItem, TextContentItem
+
+    All required parameters must be populated in order to send to server.
+
+    :ivar type: The discriminated object type. Required. Default value is None.
+    :vartype type: str
+    """
+
+    __mapping__: Dict[str, _model_base.Model] = {}
+    type: str = rest_discriminator(name="type")
+    """The discriminated object type. Required. Default value is None."""
+
+    @overload
+    def __init__(
+        self,
+        *,
+        type: str,
+    ): ...
+
+    @overload
+    def __init__(self, mapping: Mapping[str, Any]):
+        """
+        :param mapping: raw JSON to initialize the model.
+        :type mapping: Mapping[str, Any]
+        """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useless-super-delegation
+        super().__init__(*args, **kwargs)
+
+
+class ImageContentItem(ContentItem, discriminator="image_url"):
```

Comment:
Does this allow for inline images/data urls?

Reply:
Yes! let me add a sample for sending an image file up.

Reply:
I'll do this in a separate PR.